client/my-sites/upgrades/checkout/domain-details-form.jsx

```diff
@@ -70,29 +75,38 @@ class DomainDetailsForm extends PureComponent {
 			isDialogVisible: false,
 			submissionCount: 0,
 			phoneCountryCode: 'US',
-			registrantExtraInfo: null,
 			steps,
 			currentStep: first( steps ),
 		};
 	}
 
+
 	componentWillMount() {
 		this.formStateController = formState.Controller( {
 			fieldNames: this.fieldNames,
-			loadFunction: wpcom.getDomainContactInformation.bind( wpcom ),
+			loadFunction: this.loadFormState,
 			sanitizerFunction: this.sanitize,
 			validatorFunction: this.validate,
 			onNewState: this.setFormState,
 			onError: this.handleFormControllerError
 		} );
-
-		this.setState( { form: this.formStateController.getInitialState() } );
 	}
 
 	componentDidMount() {
 		analytics.pageView.record( '/checkout/domain-contact-information', 'Checkout > Domain Contact Information' );
 	}
 
+	componentDidUpdate( prevProps, prevState ) {
+		if ( ! isEqual( prevState.form, this.state.form ) ) {
+			this.props.updateContactDetailsCache( this.getMainFieldValues( this.state.form ) );
```

Comment:
Another reason it's over here atm -- `formState` is to React Virtual DOM as Redux state is to Browser DOM.

ie, I want to ensure the place we unidirectionally "write out" `formState` changes to Redux is as stable and authoritative as possible. This might be that place?